test/new_routes_test.go

```diff
@@ -50,6 +50,78 @@ func TestNewRouteInfoOnConnect(t *testing.T) {
 	if info.Nonce == "" {
 		t.Fatalf("Expected a non empty nonce in new route INFO")
 	}
+	// By default headers should be true.
+	if !info.Headers {
+		t.Fatalf("Expected to have headers on by default")
+	}
+}
+
+func TestNewRouteHeaderSupport(t *testing.T) {
+	srvA, srvB, optsA, optsB := runServers(t)
+	defer srvA.Shutdown()
+	defer srvB.Shutdown()
+
+	clientA := createClientConn(t, optsA.Host, optsA.Port)
+	defer clientA.Close()
+
+	clientB := createClientConn(t, optsB.Host, optsB.Port)
+	defer clientB.Close()
+
+	sendA, expectA := setupHeaderConn(t, clientA)
+	sendA("SUB foo bar 22\r\n")
+	sendA("PING\r\n")
+	expectA(pongRe)
+
+	if err := checkExpectedSubs(1, srvA, srvB); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	sendB, expectB := setupHeaderConn(t, clientB)
+	// Can not have \r\n in payload fyi for regex.
+	sendB("HPUB foo reply 12 14\r\nK1:V1,K2:V2 ok\r\n")
```

Comment:
Maybe add tests without reply without a queue, without reply but a queue.

Reply:
And maybe one that tests error conditions such as:
too few args, enough args but no + or | 
letters for a size for header and payload size.

Reply:
I did those tests for HPUB, will add for HMSG too.